### PR TITLE
Output dashboard console in Aspire console

### PR DESCRIPTION
**Pull request by JamesNK**
I'd like to get feedback on this approach.

There isn't configuration to turn this on and off yet.
I was thinking about having a log level to write, but getting that would require parsing income text. An alternative way to do this would have the log level set inside the dashboard.

image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2916)

### CODE REVIEW

**Comment by davidfowl** [comment]
Why can't we just make it show up in the dashboard instead?

File: src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                    {
+                        Console.Write("(Dashboard) ");
+                        Console.WriteLine(resolvedContent);
+                    }
+                    else
+                    {
+                        Console.Error.Write("(Dashboard) ");
+                        Console.Error.WriteLine(resolvedContent);
+                    }
+                }
```
Comment (davidfowl) [inline src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs:152]: One big problem with this appraoch is overlapping with any other log statements that could happen concurrently.

Comment (JamesNK) [reply src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs:152]: Yes. Logs from the dashboard are prefixed with `(Dashboard)` to make it clear where they're coming from.

Comment (tlmii) [reply src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs:152]: I was thinking he meant something else could sneak in between the Write and the WriteLine calls.

Comment (JamesNK) [reply src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs:152]: I see. Combining them and then writing them will fix that.

Comment (davidfowl) [reply src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs]: These should all just use ILogger so its syncrhronized.

Comment (JamesNK) [reply src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs]: The problem with doing that is we're receiving raw console output from the dashboard app. Writing that to ILogger would result in that info being duplicated or wrong. We'd need to parse out the level, category, message, and put it back into a local ILogger.

That didn't seem worth it to me so I just wrote raw console output from the dashboard out to the console in the apphost.

**Comment by mitchdenny** [comment]
Thinking about the reasons we might want to see the dashboard console output, its almost always going to be to debug an issue. At which point you can't guarantee that the dashboard is going to be functional. So having a way of routing it through to the AppHost console output is pretty good.

I'd probably look to have a completely distinct logger category and make these Debug level logs which wouldn't be visible by default.

**Comment by davidfowl** [comment]
Let’s just push these logs through a logger category in the app host. I like that idea @mitchdenny

**Pull request closed without merging**